llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -705,7 +707,25 @@ def fsdp_wrap_fn(self, module: nn.Module) -> bool:
 
     # Activation Checkpointing
     def activation_checkpointing_fn(self, module: nn.Module) -> bool:
-        return isinstance(module, MPTBlock)
+        if not hasattr(self.config, 'activation_checkpointing_target'):
+            return isinstance(module, MPTBlock)
+        act_ckpt_str = self.config.activation_checkpointing_target
+        act_ckpt_lst = act_ckpt_str.replace(' ', '').split(',')
+        if act_ckpt_lst:
+            if 'MPTBlock' in act_ckpt_lst or 'mptblock' in act_ckpt_lst:
+                act_ckpt_lst = ['MPTBlock']
+            for mod_name in act_ckpt_lst:
+                if mod_name.lower() == 'mptblock':
+                    mod_type = MPTBlock
+                elif mod_name in ATTN_CLASS_REGISTRY:
+                    mod_type = ATTN_CLASS_REGISTRY[mod_name]
+                elif mod_name in FFN_CLASS_REGISTRY:
+                    mod_type = FFN_CLASS_REGISTRY[mod_name]
+                elif mod_name in NORM_CLASS_REGISTRY:
+                    mod_type = NORM_CLASS_REGISTRY[mod_name]
+                else:
+                    continue
+                return isinstance(module, mod_type)
```

Comment:
this wont work on a list will it?

u might need to do something like
```suggestion
            mod_types = []
            for mod_name in act_ckpt_lst:
                if mod_name.lower() == 'mptblock':
                    mod_types += [MPTBlock]
                elif mod_name in ATTN_CLASS_REGISTRY:
                    mod_types += [ATTN_CLASS_REGISTRY[mod_name]]
                elif mod_name in FFN_CLASS_REGISTRY:
                    mod_types += [FFN_CLASS_REGISTRY[mod_name]]
                elif mod_name in NORM_CLASS_REGISTRY:
                    mod_types += [NORM_CLASS_REGISTRY[mod_name]]
            return isinstance(module, mod_types)
```

this way `activation_checkpointing_target` can actually be a list.

Reply:
they are equivalent i think, the return is inside the for loop, once we found a match, then we can return

Reply:
way I'm seeing it:

if you pass in a list thats say: `[ATTN_CLASS_REGISTRY, NORM_CLASS_REGISTRY]`
in your version, we'll only ever get `return isinstance(module, ATTN_CLASS_REGISTRY)` (even if `isinstance(module, NORM_CLASS_REGISTRY)`

in my version, you'll actually get `isinstance(module, [ATTN_CLASS_REGISTRY,, NORM_CLASS_REGISTRY])`

Reply:
I see, you are right, thanks!